src/struct_command_manager.ts

```diff
@@ -339,13 +395,26 @@ export class StructCommandManager {
     splice(start_pos: number, amt_to_remove: number) {
         console.log(start_pos + " " + amt_to_remove)
 
+        this.curr_speech = [""];
+
         var copiedStructCommand = this.deepCopyStructCommand()
         var copiedSpeechHist = this.deepCopySpeechHist()
         var oldIdx = this.curr_index;
 
-        if (this.curr_index < amt_to_remove) this.curr_index = start_pos;
-        else this.curr_index -= amt_to_remove;
-
+        var lostCursor = false;
```

Comment:
haha what isn't this code ive written. but why is it just added now

Reply:
I dont know lol I didnt do anything to this function